core/Test/Tasty/Core.hs

```diff
@@ -303,16 +308,41 @@ after deptype s =
     Nothing -> error $ "Could not parse pattern " ++ show s
     Just e -> after_ deptype e
 
+-- | Run given 'TestTree's one after another.
+--
+-- @since 1.5
+afterTree
+  :: DependencyType -- ^ whether to run the tests even if some of the dependencies fail
+  -> TestTree -- ^ dependencies
+  -> TestTree -- ^ tests to run after all dependencies have finished
+  -> TestTree
+afterTree = AfterTree
+
+-- | Run given 'TestTree's one after another.
+--
+-- @since 1.5
+sequentialTestGroup
+  :: TestName -- ^ Test group name
+  -> DependencyType -- ^ whether to run the tests even if some of the dependencies fail
+  -> [TestTree] -- ^ tests to run sequentially
+  -> TestTree
+sequentialTestGroup groupName depType =
+  testGroup groupName . maybeToList . go
+ where
+  go [] = Nothing
+  go (t:ts) = Just (foldl' (afterTree depType) t ts)
+
 -- | An algebra for folding a `TestTree`.
 --
 -- Instead of constructing fresh records, build upon `trivialFold`
 -- instead. This way your code won't break when new nodes/fields are
 -- indroduced.
 data TreeFold b = TreeFold
-  { foldSingle :: forall t . IsTest t => OptionSet -> TestName -> t -> b
+  { foldSingle :: forall t . IsTest t => OptionSet -> ExactPath -> TestName -> t -> b
```

Comment:
Ah, I see. The breaking change is in a record field. This is nasty, unlikely one can hide it beneath `PatternSynonyms`. 

Changing `TreeFold` is likely to break all third-party `tasty` ingredients.

Reply:
It would. I think we can make this backwards compatible by defining `foldSingleExact` that takes the original `foldSingle` as an argument.

But... I don't actually think it's worth it. Patching any existing code would be a matter of ignoring one (more) argument / fairly mechanical, which seems like a fair trade-off IMO.